userspace/sysdig/sysdig.cpp

```diff
@@ -72,19 +73,19 @@ static void signal_callback(int signal)
 	if(g_plugin_input)
 	{
 		//
-		// Input plugins can get stuck at any point.
+		// Input plugins can get stuck in next().
 		// When we are using one, check again in few seconds and force a quit
 		// if we are stuck.
 		//
-		if(g_terminate == true)
+		if(g_terminate == true && g_terminating == false)
```

Comment:
If i undestand the code correctly:
* if `signal_callback` is called while a plugin is inside `next`, we set `g_terminate` to true, and wait 3 more seconds
* then, if the plugin is still frozen inside `next`, we will `exit` the program (because `g_terminating` will still be false)

* If, instead, during those 3 seconds, we are now in `g_terminating` mode (ie: we are closing the live capture), what is supposed to happen here?
We will have both `g_terminate` and `g_terminating` true, and we will loop on `alarm(3);`, right?

Perhaps i am missing some pieces here :)

Reply:
correct. The idea is:
- if g_terminating doesn't get set, it means a plugin (or, more generally, the reader) is stuck on open() or next(). We assume such behavior is not good and in that case we exit   
- if g_terminating is set, it means that the plugin is working on the close. We assume it's well behaved and it will eventually return (cleanup can take time) and we just let it go forver

Reply:
I see now, thanks!